File: Doodream/큰 수 만들기.js
function solution(number, k) {
    var stack = [];
    var answer = '';

    for (let i = 0; i < number.length; i++) {
        var max = number[i];
        while (k > 0 && stack[stack.length - 1] < max) {
            stack.pop();
            k--;
        }
        stack.push(max);
    }

    stack.splice(stack.length - k, k);
    answer = stack.join('');
    return answer;
}

var number = "4177252841";
var k = 4;
console.log(solution(number, k));

File: yeonkyung/큰 수 만들기.py
def solution(number, k):
    stack = []
    for n in list(number):
        n = int(n)

        while stack:
            if k == 0:
                break
            if stack[-1] < n:
                stack.pop()
                k -= 1
            else:
                break
        stack.append(n)

    while k > 0:
        stack.pop()
        k -= 1

    return ''.join(map(str, stack))

if __name__ == "__main__":
    number = "44"
    k = 1
    print(solution(number, k))